src/main/java/org/citopt/connde/service/testing/TestReport.java

```diff
@@ -0,0 +1,627 @@
+package org.citopt.connde.service.testing;
+
+
+import com.itextpdf.text.*;
+import com.itextpdf.text.pdf.PdfWriter;
+import org.citopt.connde.domain.rules.Rule;
+import org.citopt.connde.domain.testing.TestDetails;
+import org.citopt.connde.repository.RuleRepository;
+import org.citopt.connde.repository.TestDetailsRepository;
+import org.citopt.connde.service.receiver.ValueLogReceiver;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import com.itextpdf.text.pdf.PdfPCell;
+import com.itextpdf.text.pdf.PdfPTable;
+
+import java.io.*;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+
+/**
+ * The component TestReport is used for the creation of test reports for tests of applications using the testing-tool
+ */
+@Component
+public class TestReport {
+
+
+    @Autowired
+    private TestDetailsRepository testDetailsRepository;
+
+    @Autowired
+    TestReport testEngine;
+
+    @Autowired
+    ValueLogReceiver valueLogReceiver;
+
+    @Autowired
+    RuleRepository ruleRepository;
+
+    // Date formatter
+    final String datePattern = "dd-MM-yyyy HH:mm:ss";
+    final SimpleDateFormat simpleDateFormat = new SimpleDateFormat(datePattern);
+
+    /**
+     * Generates the Test-Report with the Chart of the simulated Values and other important informations for the user.
+     *
+     * @param testId id of the specific test
+     * @param rulesBefore detail information of the selected rules before the test
+     * @return path where the TestReport can be found
+     * @throws Exception
+     */
+    public String generateTestreport(String testId, java.util.List<Rule> rulesBefore) throws Exception {
+        TestDetails test = testDetailsRepository.findById(testId);
+        String amountEvents;
+        String amountOutliers;
+        String simTime;
+        String startTestTime;
+        String endTestTime;
+
+        Document doc = new Document();
+
+        // Create a new pdf, which is named with the ID of the specific test
+        File tempFile = new File(testId + ".pdf");
+        if (tempFile.exists() && tempFile.isFile()) {
+            tempFile.delete();
+        }
+        File testReport = new File(testId + ".pdf");
+        String path = testReport.getAbsolutePath();
+
+        FileOutputStream pdfFileout = new FileOutputStream(testReport);
```

Comment:
Would it maybe be possible to store the file somewhere else, e.g. in a database or to generate the report on demand without creating a file? Storing files on disk makes the server stateful and breaks the scalability of the application.
In addition, it looks like this file is never deleted after creation, leading to a full disk at some point in time.